Mathlib/Tactic/Algebraize.lean

```diff
@@ -0,0 +1,264 @@
+will help the `algebraize` tactic access the corresponding `Algebra` property.
+
+There are two cases for what declaration corresponding to this `Name` can be.
+1. An inductive type (i.e. the `Algebra` property itself), in this case it is assumed that the
+`RingHom` and the `Algebra` property are definitionally the same, and the tactic will construct the
+`Algebra` property by giving the `RingHom` property as a term.
+
+2. A constructor for the `Algebra` property. In this case it is assumed that the `RingHom` property
+is the last argument of the constructor, and that no other explicit argument is needed. The tactic
+then constructs the `Algebra` property by applying the constructor to the `RingHom` property.
+
+Finally, if no argument is provided to the `algebraize` attribute, it is assumed that the tagged
+declaration has name `RingHom.Property` and that the corresponding `Algebra` property has name
+`Algebra.Property`. The attribute then returns `Algebra.Property` (so assume case 1 above). -/
+initialize algebraizeAttr : ParametricAttribute Name ←
+  registerParametricAttribute {
+    name := `algebraize,
+    descr :=
+"Tag that lets the `algebraize` tactic know which `Algebra` property corresponds to this `RingHom`
+property.",
+    getParam := algebraizeGetParam }
+
+end Lean.Attr
+
+namespace Mathlib.Tactic
+
+namespace Algebraize
+
+/-- Given an expression `f` of type `RingHom A B` where `A` and `B` are commutative semirings,
+this function adds the instance `Algebra A B` to the context (if it does not already exist).
+
+This function also requries the type of `f`, given by the parameter `ft`. The reason this is done
+(even though `ft` can be inferred from `f`) is to avoid recomputing `ft` in the `algebraize` tactic,
+as when `algebraize` calls `addAlgebraInstanceFromRingHom` it has already computed `ft`. -/
+def addAlgebraInstanceFromRingHom (f ft : Expr) : TacticM Unit := withMainContext do
+  let (_, l) := ft.getAppFnArgs
+  -- The type of the corresponding algebra instance
+  let alg ← mkAppOptM `Algebra #[l[0]!, l[1]!, none, none]
+  try let _ ← synthInstance alg -- If the instance already exists, we do not do anything
+  catch _ => liftMetaTactic fun mvarid => do
+    let nm ← mkFreshUserName `algInst
+    let mvar ← mvarid.define nm alg (← mkAppM `RingHom.toAlgebra #[f])
+    let (_, mvar) ← mvar.intro1P
+    return [mvar]
+
+/-- Given an expression `g.comp f` which is the composition of two `RingHom`s, this function adds
+the instance `IsScalarTower A B C` to the context (if it does not already exist). -/
+def addIsScalarTowerInstanceFromRingHomComp (fn : Expr) : TacticM Unit := withMainContext do
+  let (_, l) := fn.getAppFnArgs
+  let tower ← mkAppOptM `IsScalarTower #[l[0]!, l[1]!, l[2]!, none, none, none]
+  try let _ ← synthInstance tower -- If the instance already exists, we do not do anything
+  catch _ => liftMetaTactic fun mvarid => do
+    let nm ← mkFreshUserName `scalarTowerInst
```

Comment:
```suggestion
    let nm ← mkFreshBinderNameForTactic `scalarTowerInst
```

This ensures that one can use `set_option tactic.hygienic false`. (and also in the other 3 places this is used)